src/audio/pipeline/pipeline-stream.c

```diff
@@ -268,24 +270,123 @@ static void pipeline_trigger_xrun(struct pipeline *p, struct comp_dev **host)
 	} while (true);
 }
 
+#if CONFIG_KCPS_DYNAMIC_CLOCK_CONTROL
+static int add_pipeline_cps_consumption(struct comp_dev *current,
```

Comment:
how to deal with peak pipeline consumption? it normally 3 times than average.
avg 1275 peak 3891

Reply:
@btian1 It's a good question, but I think the only reasonably solution is to optimize the peaks out (both the absolute peak value, and the frequency they occur). We do have buffering in SRAM to survive some spikes in scheduling. But obviously if the peaks are too high and/or too frequent, this won't work. This is a bug really then. You need to either fix the code that caues the peak, or adjust the CPS data in manifest.